testing/helpers/qunitExtensions.js

```diff
@@ -363,176 +366,121 @@
+                });
+            },
+            clear: function() {
+                rules = [];
+            },
+            shouldIgnore: function(timerInfo) {
+                let skip = false;
+
+                rules.forEach(function(rule) {
+                    if(rule(timerInfo)) {
+                        skip = true;
+                        return false;
+                    }
+                });
+
+                return skip;
+            }
+        };
+    })();
+
     QUnit.timersDetector = {
-        spyWindowMethods: spyWindowMethods
+        spyWindowMethods: spyWindowMethods,
+        ignoreRules: ignoreRules
     };
 
     if(!QUnit.urlParams['notimers']) {
         return;
     }
 
-    const suppressLogOnTest = function() {
-        return /Not cleared timers detected/.test(QUnit.config.current.testName);
-    };
-
     const log = spyWindowMethods();
 
-    QUnit.timerIgnoringCheckers = (function() {
-        let checkers = [];
-        let checkersToUnregister = [];
-
-        const register = function() {
-            Array.prototype.push.apply(checkers, arguments);
-        };
-
-        const unregister = function() {
-            Array.prototype.push.apply(checkersToUnregister, arguments);
-        };
-
-        const unregisterSingle = function(checker) {
-            const index = checkers.indexOf(checker);
-
-            checkers.splice(index, 1);
-        };
-
-        const applyUnregister = function() {
-            checkersToUnregister.forEach(unregisterSingle);
-            checkersToUnregister = [];
-        };
+    ignoreRules.register(function isThirdPartyLibraryTimer(timerInfo) {
+        const callback = String(timerInfo.callback).replace(/\s/g, '');
+        const timerType = timerInfo.timerType;
 
-        const clear = function() {
-            checkers = [];
-            checkersToUnregister = [];
-        };
+        if(timerType === 'timeouts') {
+            if(
+                callback.indexOf('.Deferred.exceptionHook') > -1 || // NOTE: jQuery.Deferred are now asynchronous
+                callback.indexOf('e._drain()') > -1 // NOTE: SystemJS Promise polyfill
+            ) {
+                return true;
+            }
 
-        const needSkip = function(timerInfo) {
-            let skip = false;
+            if(
+                window.navigator.userAgent.indexOf('Edge/') > -1 && // NOTE: Native Promise in Edge
+                callback.indexOf('function(){[nativecode]}') > -1
+            ) {
+                return true;
+            }
+        }
+    });
 
-            checkers.forEach(function(checker) {
-                if(checker(timerInfo)) {
-                    skip = true;
-                    return false;
-                }
-            });
+    const logTestFailure = function(timerInfo) {
+        const timeoutString = timerInfo.timeout ? '\nTimeout: ' + timerInfo.timeout : '';
 
-            return skip;
-        };
+        const message = [
```

Comment:
Why don't use a multiline string?
```
        const message = `Not cleared timer detected.
Timer type: ${timerInfo.timerType}
Id: ${timerInfo.timerId}
Callback:
${timerInfo.callback}
${timeoutString}`;
```

Reply:
Interpolated strings are not supported in IE:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals